model-optimizer/mo/back/offline_transformations.py

```diff
@@ -37,10 +37,42 @@ def compress_model(func: object):
     from openvino.offline_transformations_pybind import compress_model_transformation  # pylint: disable=import-error,no-name-in-module
     compress_model_transformation(func)
 
-def apply_offline_transformations(input_model: str, framework: str, transforms: list, compress_fp16=False):
+
+def add_layouts(ov_function, argv: argparse.Namespace):
+    from openvino.runtime import Layout  # pylint: disable=import-error,no-name-in-module
+    layout_values = argv.layout_values
+    if '' in layout_values:
+        if len(ov_function.inputs) > 1:
+            input_names = [list(ov_input.get_tensor().get_names())[0] for ov_input in ov_function.inputs]
+            raise Error('Layout without name can be specified for models with only one input, '
+                        'but provided model has {} inputs: \'{}\'. '
+                        'Please specify explicitly input/output name for --layout option'
+                        .format(len(input_names), input_names))
+        layout_values = {
+            list(ov_function.input().get_tensor().get_names())[0]: {
+                'source_layout': layout_values[''].get('source_layout'),
+                'target_layout': layout_values[''].get('target_layout')
+            }
+        }
+    set_layout_names = set(layout_values.keys())
+    for idx, ov_input in enumerate(ov_function.inputs):
+        found = set.intersection(set(ov_input.get_tensor().get_names()), set_layout_names)
+        if len(found) == 1:
+            source_layout = layout_values[list(found)[0]]['source_layout']
+            if source_layout:
+                ov_function.get_parameters()[idx].layout = Layout(source_layout)
+    for idx, ov_output in enumerate(ov_function.outputs):
+        found = set.intersection(set(ov_output.get_tensor().get_names()), set_layout_names)
+        if len(found) == 1:
```

Comment:
What if `len(found) != 1`? Is it possible?

Reply:
Same, if nothing was found go to next iteration. Multiple names shouldn't be possible.